float32[] line_lengths
